pkg/vmimage/vmimage.go

```diff
@@ -114,8 +131,85 @@ func (builder *Builder) generateTemplate() (map[string]interface{}, error) {
 	return template, nil
 }
 
+func (builder *Builder) vmImageReference() *compute.ImageReference {
+	if !builder.Validate {
+		// Building a new image in a VM
+		return &compute.ImageReference{
+			Publisher: to.StringPtr("RedHat"),
+			Offer:     to.StringPtr("RHEL"),
+			Sku:       to.StringPtr("7-RAW"),
+			Version:   to.StringPtr("latest"),
+		}
+	}
+
+	// Validating an existing marketplace image
+	if builder.hasMarketplaceVMImageRef() {
+		return &compute.ImageReference{
+			Publisher: to.StringPtr(builder.ImagePublisher),
```

Comment:
suggestion/discussion: 
I know hard coding is bad but should we verify that the image reference is our exact one?

Reply:
I would like to avoid hardcoding SKU and version, because I guess we will have different SKU and versions for v4 openshift, but we can hardcode the publisher and offer. However I do not see why we would want to do this. Do you have any reasons in mind?

Reply:
If the variables are tied to our publishing image then this prevents fat fingers and guarantees our publishing information is set. These will hopefully never change.  Just a suggestion.

I will not block this PR because of this.

Reply:
To guarantee that publisher and offer is set I added this info into `hack/vmimage-validate.sh` with ability to override, if needed:

```
IMAGE_PUBLISHER="${IMAGE_PUBLISHER:-redhat}"
IMAGE_OFFER="${IMAGE_OFFER:-osa}"
```

But taking into account that this is not going to change often (likely to never change), I think we can simplify my changes by removing these variables from the validation script, removing flags from the go command and just hardcode publisher and offer names here.

I'm happy to do this.